projects/core/src/cms/occ/occ-cms-component.adapter.ts

```diff
@@ -40,26 +40,52 @@ export class OccCmsComponentAdapter implements CmsComponentAdapter {
     pageSize = ids.length,
     sort?: string
   ): Observable<CmsComponent[]> {
-    const requestParams = this.getComponentsRequestParams(
+    const idList: IdList = { idList: ids };
+
+    let requestParams = this.getComponentsRequestParams(
+      idList,
       pageContext,
+      true,
       currentPage,
       pageSize,
       sort
     );
 
-    const idList: IdList = { idList: ids };
-
     return this.http
-      .post<CmsComponentList>(
+      .get<CmsComponentList>(
         this.getComponentsEndpoint(requestParams, fields),
-        idList,
         {
           headers: this.headers,
         }
       )
       .pipe(
         pluck('component'),
-        this.converter.pipeableMany(CMS_COMPONENT_NORMALIZER)
+        this.converter.pipeableMany(CMS_COMPONENT_NORMALIZER),
+        catchError(error => {
+          if (error.status === 400) {
+            requestParams = this.getComponentsRequestParams(
```

Comment:
Can we move this part into a new function? So, we have 2 `loadList` functions (with different names). So, it is consistent with backend. If the "post" endpoint is removed in future, it is easy to remove one function from SPA. And the codes look more clear. No need to modify `getComponentsRequestParams`. `requestParams['componentIds']=xxx` can just be added into the `get loadList` function.